src/modules/tab_completion/index.js

```diff
@@ -109,6 +116,27 @@ class TabCompletionModule {
         } else if (keyCode !== keyCodes.Shift) {
             this.tabTries = -1;
         }
+
+        // Message history
+        if (keyCode === keyCodes.UpArrow) {
```

Comment:
Do we? It's pretty unobtrusive. Unlike the old one, it only goes to the previous if you're at the start of your message, and it also saves the edits you made the messages, so when you go back, you don't lose any data. The chances of it accidentally fucking you over are basically non-existent.